neo/Network/P2P/TaskManager.cs

```diff
@@ -266,5 +271,13 @@ internal protected override bool IsHighPriority(object message)
                     return false;
             }
         }
+
+        internal protected override bool ShallDrop(object message, IEnumerable queue)
+        {
+            if (!(message is TaskManager.NewTasks tasks)) return false;
+            // Remove duplicate tasks
+            if (queue.OfType<TaskManager.NewTasks>().Any(x => x.Payload.Type == tasks.Payload.Type && x.Payload.Hashes.SequenceEqual(tasks.Payload.Hashes))) return true;
```

Comment:
Are these duplicate messages coming from the same node or from different nodes?

Reply:
Different nodes.

Reply:
Maybe we should limit the number of connections? It looks like a performance issue, not a logic issue.

Reply:
I think that we need to re-run the tests after the last fix, @yongjiema, because the FifoSet had some problems in the ExceptWith command.

But I agree with the last comment I saw from you guys, @KickSeason, about the TaskCompleted. It becomes each time harder to really complete a task after it passes through ProtocolHandle InvFunction.

Reply:
> I think that we need to re-run the tests after the last fix, @yongjiema, because the FifoSet had some problems in the ExceptWith command.
> 
> But I agree with the last comment I saw from you guys, @KickSeason, about the TaskCompleted. It becomes each time harder to really complete a task after it passes through ProtocolHandle InvFunction.

The test did in my local already included FIFOSet fix before preparing this PR.

Reply:
The fix was wrong with a "minor" problem..,heuahahsu

Reply:
May I know what's the "minor" problem?

Reply:
As I said below, the HashSet that coordinates the `KnowHashes` use function `ExceptWith` for filtering and this function was not working probably on the new `FifoSet`. The last commit probably solved that, we also created some UT for being sure about that.

This surely could cause bad behavior.

Reply:
Oh, I see, I have already noticed the problem, below is some part of my local FIFOSet during the testing before this PR:

```cs
// FIFOSet class

public IEnumerator<T> GetEnumerator()
{
    return data.Keys.OfType<T>().GetEnumerator();
}

IEnumerator IEnumerable.GetEnumerator()
{
    return data.Keys.GetEnumerator();
}
```

Reply:
I think that this was resolved, right? It was just a discussion, correct?